### DISCUSSION TITLE
Netcdf Support

### DISCUSSION THREAD

**Discussion by mgorfer** [nushell/nushell#10758] [Ideas]
Are there any plans to add support for netcdf files?
In my main job, I have to deal with netcdf files all day.
There are some options to read these files of course, like `ncview`, 
but most of the time I just open a python terminal and use python modules to interact with these files.

Having similar features as there are for `squlite` databases available in nushell for netcdf files would be a game changer for me.

**Comment by fdncred** [comment]
what is `netcdf`?

**Reply by mgorfer** [reply]
Netcdf is a data format.
https://www.unidata.ucar.edu/software/netcdf/

It is widely used in the scientific community, and there is good support for popular programming languages like [C](https://docs.unidata.ucar.edu/netcdf-c/current/), [C++](https://docs.unidata.ucar.edu/netcdf-cxx/current/), [Java](https://www.unidata.ucar.edu/software/netcdf-java/), and [Fortran](https://docs.unidata.ucar.edu/netcdf-fortran/current/), Python, IDL, MATLAB, R, Ruby, and Perl.

I am not sure about the popularity of this file format outside my field, or how difficult it would be to add support for Nushell.
There is also support for Netcdf in Rust: https://docs.rs/netcdf/latest/netcdf/

**Reply by fdncred** [reply]
Sounds like a good thing for a plugin. It looks like there are some Rust crates for it too https://crates.io/search?q=netcdf

**Reply by mgorfer** [reply]
I would be very interested in trying to build a basic netcdf plugin for core functionality.

I am not an experienced Rust developer yet, though. 
Do you think it would still be feasible for someone with lots of experience with python? 
Or better to look for another project to get more proficient in Rust first?

**Reply by fdncred** [reply]
I think it's feasible to write a rust plugin for nushell without having a ton of rust experience. My suggestion would be to:
1. install and use `cargo-generate` with this url https://github.com/fdncred/nu_plugin_template
2. look at the awesome-nu repo because there are a bunch of already written rust plugins to look at https://github.com/nushell/awesome-nu#plugins
3. look at our sample plugins in the nushell repo nu-plugin-example and maybe nu-plugin-python
4. follow along with this chapter in our hidden contributor book https://www.nushell.sh/contributor-book/plugins.html

**Reply by ghost** [reply]
in case you don't manage to create something in rust you can also write a nu plugin in python or any other language - the code is just a lot more verbose compared to a implementation in rust. a example for a python nu plugin: <https://github.com/nushell/nushell/blob/main/crates/nu_plugin_python/nu_plugin_python_example.py>. (the filename has to start with `nu_plugin_` and be executable. it can be registered with `register nu_plugin_SOMETHING.py` like any other plugin)
